Lib/pathlib.py

```diff
@@ -1091,7 +1091,8 @@ def glob(self, pattern):
             raise NotImplementedError("Non-relative patterns are unsupported")
         selector = _make_selector(tuple(pattern_parts))
         for p in selector.select_from(self):
-            yield p
+            if not pattern.endswith(self._flavour.sep) or p.is_dir():
```

Comment:
It would be more efficient to not check whether the pattern ends with a component separator at every iteration.
```python
dironly = ...
for p in selector.select_from(self):
    if not dironly or p.is_dir():
        yield p
```
or
```python
if ...:
    for p in selector.select_from(self):
        if p.is_dir():
            yield p
else:
    yield from selector.select_from(self):
```
It may be also worth to pass the `dironly` flag to `select_from()` and perform filtering internally. This may be more efficient, and the existing code already supports filtering directories only.

Reply:
Thank you for your advice.
I'm not sure how to use `_Selector.select_from`, so I chose the second one.